packages/components/src/entities/utils.tsx

```diff
@@ -176,6 +176,31 @@ export function getSampleDeleteMessage(canDelete: boolean, deleteInfoError: bool
     return deleteMsg;
 }
 
+export const getDataClassTemplateUrl = (
+    queryInfo: QueryInfo,
+    exportConfig: any = {},
+    excludeColumns: string[] = ['flag', 'alias', 'lsid', 'Ancestors'],
+): string => {
+    const schemaQuery = queryInfo.schemaQuery;
+    if (!schemaQuery) return undefined;
+
+    // Issue 46593: if the table XML metadata override specifies a custom importTemplate, use it
+    if (queryInfo.importTemplates?.[0]?.url.toLowerCase().indexOf('exportexceltemplate.view') === -1) {
+        return queryInfo.importTemplates?.[0].url;
```

Comment:
nit: No need for a defensive `?` check here since it has already been verified in the if statement.

Reply:
Done.